test/ussd_registration.test.js

```diff
@@ -554,6 +559,97 @@ describe("Mama Nigeria App", function() {
+                    // test data
+                    var testDate = moment("2015-07-26");
+                    var limit = 7;     // should determine the size of the returned array
+                    var increment = -1; // should determine subsequent direction of array elements
+
+                    // function call
+                    var expectedChoiceArray = go.utils.make_month_choices($, testDate, limit, increment);
+
+                    // expected results
+                    assert.equal(expectedChoiceArray.length, limit);
+                    assert.equal(expectedChoiceArray[0].value, "072015");
+                    assert.equal(expectedChoiceArray[1].value, "062015");
+                    assert.equal(expectedChoiceArray[2].value, "052015");
+                    assert.equal(expectedChoiceArray[3].value, "042015");
+                    assert.equal(expectedChoiceArray[4].value, "032015");
+                    assert.equal(expectedChoiceArray[5].value, "022015");
+                    assert.equal(expectedChoiceArray[6].value, "012015");
+                });
+                it('should return a Choice array of correct size - forward across years', function() {
+                    // test data
+                    var testDate = moment("2015-12-26");
+                    var limit = 4;     // should determine the size of the returned array
+                    var increment = 1; // should determine subsequent direction of array elements
+
+                    // function call
+                    var expectedChoiceArray = go.utils.make_month_choices($, testDate, limit, increment);
+
+                    // expected results
+                    assert.equal(expectedChoiceArray.length, limit);
+                    assert.equal(expectedChoiceArray[0].value, "122015");
+                    assert.equal(expectedChoiceArray[1].value, "012016");
+                    assert.equal(expectedChoiceArray[2].value, "022016");
+                    assert.equal(expectedChoiceArray[3].value, "032016");
+                });
+                it('should return an array of choices - backwards across years', function() {
+                    // test data
+                    var testDate = moment("2015-01-26");
+                    var limit = 3;     // should determine the size of the returned array
+                    var increment = -1; // should determine subsequent direction of array elements
+
+                    // function call
+                    var expectedChoiceArray = go.utils.make_month_choices($, testDate, limit, increment);
+
+                    // expected results
+                    assert.equal(expectedChoiceArray.length, limit);
+                    assert.equal(expectedChoiceArray[0].value, "012015");
+                    assert.equal(expectedChoiceArray[1].value, "122014");
+                    assert.equal(expectedChoiceArray[2].value, "112014");
+                });
+                //
```

Comment:
Please add a test with `limit = 3, increment = 3` just to illustrate what happens.